tedge/src/cli/connect/command.rs

```diff
@@ -176,7 +184,40 @@ where
 // the check can finish in the second try as there is no error response in the first try.
 
 #[tokio::main]
-async fn check_connection_c8y(port: u16) -> Result<(), ConnectError> {
+async fn check_connection_c8y(port: u16) -> Result<DeviceStatus, ConnectError> {
+    for i in 0..2 {
+        print!("Try {} / 2: Sending a message to Cumulocity. ", i + 1,);
+
+        let create_device_result = create_device(port);
+        match create_device_result.await {
```

Comment:
Even for simple connection check seems like the create_device will be called, isn't it?

Reply:
Yup. This is how we've been doing the connection check all this while and I've just refactored it into a separate method. It was done this way due to the absence of another SmartREST message that can give us that device existence information. Once we switch to HTTP requests in future for these device management requests, it'd be better.

Reply:
>  Once we switch to HTTP requests in future for these device management requests, it'd be better.

If we switch to HTTP here, we will have to find another way to check the MQTT connection. By the way the MQTT connection is required to get a JWT token for HTTP.

Reply:
Doing connection check by retrieving JWT token seems like a good idea. We could do that in a follow-up task.